agent/engine/dependencygraph/graph.go

```diff
@@ -339,6 +336,46 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 	}
 }
 
+func verifyShutdownOrder(target *apicontainer.Container, existingContainers map[string]*apicontainer.Container) error {
+	// If the target is already terminal or if it doesn't need to be terminal, shutdown order doesn't apply
+	if !target.DesiredTerminal() || target.KnownTerminal() {
+		return nil
+	}
+
+	// Considered adding this to the task state, but this will be at most 45 loops,
+	// so we err'd on the side of having less state.
+	missingShutdownDependencies := []string{}
+
+	for _, existingContainer := range existingContainers {
+		for _, dependency := range existingContainer.DependsOn {
+			// If another container declares a dependency on our target, we will want to verify that the container is
+			// stopped.
+			if dependency.Container == target.Name {
+				if !existingContainer.KnownTerminal() {
+					missingShutdownDependencies = append(missingShutdownDependencies, existingContainer.Name)
+				}
+			}
+		}
+	}
+
+	if len(missingShutdownDependencies) == 0 {
+		return nil
+	}
+
+	return fmt.Errorf("dependency graph: target %s needs other containers stopped before it can stop: [%s]",
+		target.Name, strings.Join(missingShutdownDependencies, "], ["))
+}
+
+func verifyContainerOrderingStatus(dependsOnContainer *apicontainer.Container) bool {
```

Comment:
Is this function part of your commit? A bit confused

Reply:
I did a bad rebase -- will fix